dll/win32/imm32/imm.c

```diff
@@ -2609,23 +2609,29 @@ BOOL WINAPI ImmIsUIMessageW(
 BOOL WINAPI ImmNotifyIME(
   HIMC hIMC, DWORD dwAction, DWORD dwIndex, DWORD dwValue)
 {
-    InputContextData *data = get_imc_data(hIMC);
+    DWORD idImeThread, idThread;
+    HKL hKL;
+    PIMEDPI pImeDpi;
+    BOOL ret;
 
-    TRACE("(%p, %d, %d, %d)\n",
-        hIMC, dwAction, dwIndex, dwValue);
+    TRACE("ImmNotifyIME(%p, %d, %d, %d)\n", hIMC, dwAction, dwIndex, dwValue);
 
-    if (hIMC == NULL)
+    if (hIMC)
     {
-        SetLastError(ERROR_SUCCESS);
-        return FALSE;
+        idImeThread = Imm32QueryWindow(hIMC, 1);
```

Comment:
"1": magic value? (see this list: https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/include/ntuser.h;hb=9a42267b06861a057163729ca188438d157f13b8#l2841 )

Reply:
Used `QUERY_WINDOW_...`. OK.